### PR TITLE
`dashboard-console`: fix wonky scrollback / nav behavior

### PR DESCRIPTION
**Pull request by snickell**
We had hacked color into `bin/dashboard-console` using a method that resulted in broken arrow navigation due to not closing color codes. Additionally, it did not keep a consistent indent in history navigation, making it confusing and hard to follow.

This uses IRB's builtin colorize methods to properly terminate color codes, and uses more normal IRB indent "codes".

Based on the technique used for the new default rails console formatting (merged jan 19): https://github.com/rails/rails/pull/50796/files, but adds our app name in as well.

Before, broken arrow-to-navigate in `bin/dashboard-console` results in wild cursor positions:

https://github.com/code-dot-org/code-dot-org/assets/223277/4a1848a5-62bc-4533-968f-2d07e0058e21

After, arrow-to-navigate is rock solid:

https://github.com/code-dot-org/code-dot-org/assets/223277/0781a3d1-739d-43e1-9213-5cb9606a0ce8

Before (inconsistent indenting, weird history behavior):
image

After (consistent indenting of history):
image

### CODE REVIEW

**Comment by snickell** [comment]
https://github.com/code-dot-org/code-dot-org/assets/223277/db1da9da-0922-4899-a617-43905b6fcd04

**Comment by snickell** [comment]
Before, using arrows to navigate often results in cursor drawing in invalid locations:
https://github.com/code-dot-org/code-dot-org/assets/223277/481ea922-0bef-4b20-93fc-130a1fcda8a2

After, using arrows to navigate is rock solid:
https://github.com/code-dot-org/code-dot-org/assets/223277/a3c40702-5ea4-448f-93ec-022bc7a49d43

**Comment by snickell** [comment]
Before, broken (using arrow keys to navigate results in cursor going into la-la land)
https://github.com/code-dot-org/code-dot-org/assets/223277/8fdde6c4-1b9d-4546-ac97-d3d1a235ba5f

After, fixed (using arrow keys to navigate is rock solid):
https://github.com/code-dot-org/code-dot-org/assets/223277/48f75db3-41c9-4539-9ff5-2725eeeeaf55

I believe the issue was improper closing of the terminal color escape codes.

**Comment by snickell** [comment]
https://github.com/code-dot-org/code-dot-org/assets/223277/e29e2820-15e4-4c20-ad4e-081c8994bf87

**Comment by snickell** [comment]
https://github.com/code-dot-org/code-dot-org/assets/223277/0781a3d1-739d-43e1-9213-5cb9606a0ce8


https://github.com/code-dot-org/code-dot-org/assets/223277/4a1848a5-62bc-4533-968f-2d07e0058e21

**Review by cnbrenci — APPROVED** [review]
Omg. Yes. Plz. This makes me so happy. Every dev is going to be less annoyed. Amazing. Thank you!!!! 🙏🙏🙏

**Comment by snickell** [comment]
Looks like drone failed due to an intermittent mysql connection, restarting it:
```
=====ERROR["test_find_channel_owner_returns_nil_for_channel_with_no_owner", "UserTest", 672.9893739940017]
--
410 | test_find_channel_owner_returns_nil_for_channel_with_no_owner#UserTest (672.99s)
411 | Minitest::UnexpectedError:         Sequel::DatabaseError: Mysql2::Error::TimeoutError: Timeout waiting for a response from the last query. (waited 30 seconds)
412 | test/testing/projects_test_utils.rb:31:in `with_storage_id_for'
413 | test/testing/projects_test_utils.rb:9:in `with_channel_for'
414 | test/testing/projects_test_utils.rb:35:in `with_anonymous_channel'
415 | test/models/user_test.rb:4932:in `block in <class:UserTest>'
416 | test/testing/setup_all_and_teardown_all.rb:36:in `run'
```

**Review by davidsbailey — APPROVED** [review]
Thanks for the fix Seth! These kind of issues have definitely been a pain when using a single line as well, and I'm able to confirm that this fixes the kind of issue I have been seeing.
